guarddog/cli.py

```diff
@@ -97,10 +123,92 @@ def scan(identifier, version, rules, exclude_rules, json, exit_non_zero_on_findi
         exit_with_status_code(results)
 
 
-# Determines if the input passed to the 'scan' command is a local package name
-def is_local_package(input):
-    identifier_is_path = re.search(r"(.{0,2}\/)+.+", input)
-    return identifier_is_path or input.endswith('.tar.gz')
+def _list_rules(ecosystem):
+    metadata_detectors = get_metadata_detectors(ecosystem)
+    if len(SOURCECODE_RULES[ecosystem]) > 0:
+        print("Available source code rules:")
+        for rule in SOURCECODE_RULES[ecosystem]:
+            print(f"\t{rule}")
+    if len(metadata_detectors.keys()) > 0:
+        print("Available metadata detectors:")
+        for detector in metadata_detectors.keys():
+            print(f"\t{detector}")
+
+
+@cli.group
+def npm(**kwargs):
+    """ Scan a npm package or verify a npm project
+    @param kwargs:
+    @return:
+    """
+    pass
+
+
+@cli.group
+def pypi(**kwargs):
+    """ Scan a PyPI package or verify a PyPI project
+    """
+    pass
+
+
+@npm.command("scan")
+@common_options
+@version_option
+def scan_npm(target, version, rules, exclude_rules, json, exit_non_zero_on_finding):
+    """ Scan a given npm package
+    """
+    return _scan(target, version, rules, exclude_rules, json, exit_non_zero_on_finding, ECOSYSTEM.NPM)
+
+
+@npm.command("verify")
+@common_options
+def verify_npm(target, rules, exclude_rules, json, exit_non_zero_on_finding):
+    """ Verify a given npm project
+    """
+    return _verify(target, rules, exclude_rules, json, exit_non_zero_on_finding, ECOSYSTEM.NPM)
+
+
+@pypi.command("scan")
+@common_options
+@version_option
+def scan_pypi(target, version, rules, exclude_rules, json, exit_non_zero_on_finding):
+    """ Scan a given PyPI package
+    """
+    return _scan(target, version, rules, exclude_rules, json, exit_non_zero_on_finding, ECOSYSTEM.PYPI)
+
+
+@pypi.command("verify")
+@common_options
+def verify_pypi(target, rules, exclude_rules, json, exit_non_zero_on_finding):
+    """ Verify a given Pypi project
+    """
+    return _verify(target, rules, exclude_rules, json, exit_non_zero_on_finding, ECOSYSTEM.PYPI)
+
+
+@pypi.command("list-rules")
+def list_rules_pypi():
+    """ Print available rules for PyPI
+    """
+    return _list_rules(ECOSYSTEM.PYPI)
+
+
+@npm.command("list-rules")
+def list_rules_npm():
+    """ Print available rules for npm
+    """
+    return _list_rules(ECOSYSTEM.NPM)
+
+
+@cli.command(context_settings={"ignore_unknown_options": True}, deprecated=True)
+@click.argument('target', nargs=-1)
+def verify(target):
+    exit(1)
+
+
+@cli.command(context_settings={"ignore_unknown_options": True}, deprecated=True)
+@click.argument('target', nargs=-1)
+def scan(target):
+    exit(1)
```

Comment:
Current usage shows:

```
$ guarddog scan xxx
DeprecationWarning: The command 'scan' is deprecated.
(exited with status code 1)
```

That's not so much a warning - can we proxy it to `scan_pypi` at first and only break it in a future version? What do you think?

Reply:
sure, will do.